src/deploy.js

```diff
@@ -5,17 +5,23 @@ const send = require('./send');
 
 const { ERC1820_REGISTRY_DEPLOY_TX } = require('./data');
 
-async function ERC1820Registry (from) {
+async function ERC1820Registry (funder) {
   // Read https://eips.ethereum.org/EIPS/eip-1820 for more information as to how the ERC1820 registry is deployed to
   // ensure its address is the same on all chains.
 
+  const ERC1820RegistryAddress = '0x1820b744B33945482C17Dc37218C01D858EBc714';
+
+  if ((await web3.eth.getCode(ERC1820RegistryAddress)).length > '0x'.length) {
+  	throw new Error(`An ERC1820 registry contract instance already exists at ${ERC1820RegistryAddress}`);
```

Comment:
On second thought, perhaps we just want to return the address/instance directly instead of erroring or attempting to redeploy? I'm thinking of a test suite where there are multiple test files that need to access the instance. If they all call `deploy.ERC1820Registry` it will fail, and there is no other way currently of obtaining an instance.

Incidentally this works as an argument against naming the module `deploy`. :slightly_smiling_face:

Reply:
_grumbles_ yeah, you're right